companion/src/storage/categorized.cpp

```diff
@@ -136,20 +143,33 @@ bool CategorizedStorageFormat::write(const RadioData & radioData)
     if (!writeFile(modelData, modelFilename)) {
       return false;
     }
-    if (getCurrentFirmware()->getCapability(HasModelCategories)) {
-      int categoryIndex = model.category;
-      if (currentCategoryIndex != categoryIndex) {
-        modelsList.append(QString().sprintf("[%s]\n", radioData.categories[model.category].name));
-        currentCategoryIndex = categoryIndex;
-      }
-    }
-    if (IS_HORUS(getCurrentBoard())) {
-      modelsList.append(QString("%1\n").arg(model.filename));
```

Comment:
While you at it, the original code here basically uses these two conditions:
 * `getCurrentFirmware()->getCapability(HasModelCategories)`
 * `IS_HORUS(getCurrentBoard())`

In this context the two actually mean the same. So I would loose the second condition and base all decisions in this functions on a `getCurrentFirmware()->getCapability(HasModelCategories)` This condition determines if output has either:
 * categories, models, sorted with your map<> code
 * no categories, but each model has model number.